Normalize request body ContentTypes

rayvincent2 opened the pull request:
Fixes #862 

First I tackled ensuring that any ContentType type could be normalized so we could perform easy comparisons between what an HTTP request may send vs what the openapi document is expecting.  By normalizing as well, I was able to easily ensure that cache keys for validators don't have any conflicts.

Tests have been added to showcase the working solution.

rayvincent2 commented:
Has anybody had a chance to review this?

cdimascio commented on src/middlewares/util.ts:
```diff
@@ -30,12 +36,30 @@ export class ContentType {
+    }
+    types.push(new ContentType(this.mediaType));
+
+    if (!this.parameters['charset']) {
```
Looks like this will add utf-8 as an equivalent content type 👍

cdimascio commented on src/middlewares/util.ts:
```diff
@@ -30,12 +36,30 @@ export class ContentType {
+    return types;
+  }
+
+  public normalize(excludeParams: string[] = ['boundary']) {
```
Nit: add the method’s return type

rayvincent2 replied:
Done!

cdimascio commented:
> Has anybody had a chance to review this?

Getting to this now. Looks good. Thank you! One minor review nit

rayvincent2 commented:
> > Has anybody had a chance to review this?
> 
> Getting to this now. Looks good. Thank you! One minor review nit

Thanks for the review!  I went ahead and made the update, rebased, and squashed.  It's ready for another look.
